### DISCUSSION TITLE
convert claims to convert and cp files, but some files aren't created (when using --path query)

### DISCUSSION THREAD

**Discussion by DamonLane** [beetbox/beets#4618] [General]
Hello! I've been using beets for a few years but this is my first time with the convert plugin. I'm using convert (separate from import) to create a compressed copy of my entire library for a mobile device. I had confusing results that appear to be related to whether I use a path query or not. When I run,

`beet convert -p /home/damon/Music` 

it appears to work through all the files successfully, showing that it either called ffmpeg or cp for each file, e.g.:
```
convert: cp /home/damon/Music/Yeah Yeah Yeahs/Fever to Tell/09 Maps.mp3 /home/damon/beets/converted_music_for_phone/Yeah Yeah Yeahs/Fever to Tell/09 Maps.mp3
convert: cp /home/damon/Music/Yeah Yeah Yeahs/Show Your Bones/01 Gold Lion.mp3 /home/damon/beets/converted_music_for_phone/Yeah Yeah Yeahs/Show Your Bones/01 Gold Lion.mp3
convert: cp /home/damon/Music/Yeah Yeah Yeahs/Show Your Bones/06 Cheated Hearts.mp3 /home/damon/beets/converted_music_for_phone/Yeah Yeah Yeahs/Show Your Bones/06 Cheated Hearts.mp3
```
 but when I go to the destination folder, not all the files are there, e.g. the three files shown above are missing.
 
```ls /home/damon/beets/converted_music_for_phone/Yeah*
ls: cannot access '/home/damon/beets/converted_music_for_phone/Yeah*': No such file or directory
```
If I give convert a more specific query... 
```
beet -vv convert Yeahs
user configuration: /home/damon/.config/beets/config.yaml
data directory: /home/damon/.config/beets
plugin paths: 
Sending event: pluginload
library database: /home/damon/beets/music_library
library directory: /home/damon/Music
Sending event: library_opened
convert: cp /home/damon/Music/Yeah Yeah Yeahs/Fever to Tell/09 Maps.mp3 /home/damon/beets/converted_music_for_phone/Yeah Yeah Yeahs/Fever to Tell/09 Maps.mp3
convert: cp /home/damon/Music/Yeah Yeah Yeahs/Show Your Bones/01 Gold Lion.mp3 /home/damon/beets/converted_music_for_phone/Yeah Yeah Yeahs/Show Your Bones/01 Gold Lion.mp3
convert: cp /home/damon/Music/Yeah Yeah Yeahs/Show Your Bones/06 Cheated Hearts.mp3 /home/damon/beets/converted_music_for_phone/Yeah Yeah Yeahs/Show Your Bones/06 Cheated Hearts.mp3
```
I see the same cp action, but this time the copy actually happens and those files show up in the destination:
```
ls /home/damon/beets/converted_music_for_phone/Yea*
'Fever to Tell'  'Show Your Bones'
```
When I was preparing this example, I noticed that the "Yeahs" query worked while -p /home/damon/Music/Yeah\ Yeah\ Yeahs acted like the first call for the whole library - showing that it copied the files, but not doing it. So maybe that's it. I may have used path query because I didn't know what to use to get the whole library. But I just did a plain `beets convert` and everything looks at it should. If my beets library is all in one folder, shouldn't calling the path to that folder be the same? If the query is the difference, why does the beets output look like it worked? Although I'm happy now that my conversion is complete, the output not matching the actions confused me and I spent some time converting my collection one band at a time just to get it done. I should have written this up earlier - it's amazing how often one can solve their problem while describing it in detail to ask for help!

I searched the issues, discussions, and discourse and the only thing that seems like it could be related was this:
https://discourse.beets.io/t/files-not-being-renamed/191 Now that I see my issue was a difference between queries, I don't think mine was related to this other one, but we both saw output messages that disagreed with the results after the command.

**Comment by jackwilsdon** [comment]
The `-p` flag for `convert` tells the plugin to pretend to convert without actually doing it (useful to see what the plugin is planning on doing):
```
Usage: beet convert [options]

Options:
  -h, --help            show this help message and exit
  -p, --pretend         show actions but do nothing
  -t THREADS, --threads=THREADS
                        change the number of threads,
                        defaults to maximum available processors
  -k, --keep-new        keep only the converted
                        and move the old files
  -d DEST, --dest=DEST  set the destination directory
  -f FORMAT, --format=FORMAT
                        set the target format of the tracks
  -y, --yes             do not ask for confirmation
  -l, --link            symlink files that do not
                        need transcoding.
  -H, --hardlink        hardlink files that do not
                        need transcoding. Overrides --link.
  -a, --album           match albums instead of tracks
```

If you only want to match items in a specific directory, you want to use a [path query](https://beets.readthedocs.io/en/stable/reference/query.html#path-queries).

**Comment by DamonLane** [comment]
omg, how embarrassing! I do have these in my bash history so I did try a proper path query earlier, and which didn't get my full library at the destination, but I don't have detailed notes. Probably more user error :smile: 
```
beet convert path:Music/
beet convert path:/home/damon/Music
```
